src/Docker.DotNet/Models/ServiceFilter.cs

```diff
@@ -10,24 +10,39 @@ public class ServicesListParameters
         public ServiceFilter Filters { get; set; }
     }
 
-    public class ServiceFilter : Dictionary<string, string>
+    public class ServiceFilter : Dictionary<string, string[]>
     {
-        public string Id
+        public string[] Id
         {
             get => this["id"];
             set => this["id"] = value;
         }
-        public string Label
+        public string[] Label
         {
             get => this["label"];
             set => this["label"] = value;
         }
-        public ServiceCreationMode Mode
+        public ServiceCreationMode[] Mode
         {
-            get => !Enum.TryParse(this["mode"], out ServiceCreationMode mode) ? ServiceCreationMode.Replicated : mode;
-            set => this["mode"] = value.ToString();
+            get
+            {
+                if (this["mode"].Length == 0)

```

Comment:
If this is an `array` filter I would prefer the default value is `null`.

Reply:
It is null as default now